editor/src/core/model/element-metadata-utils.ts

```diff
@@ -551,52 +550,15 @@ export const MetadataUtils = {
       }
     }
   },
-  templatePathToStaticTemplatePath(
-    metadata: Array<ComponentMetadata>,
-    path: TemplatePath | null,
-  ): StaticTemplatePath | null {
+  templatePathToStaticTemplatePath(path: TemplatePath | null): StaticTemplatePath | null {
     if (path == null || TP.isScenePath(path)) {
       return path
     } else {
-      return this.dynamicPathToStaticPath(metadata, path)
+      return this.dynamicPathToStaticPath(path)
     }
   },
-  dynamicPathToStaticPath(
-    metadata: Array<ComponentMetadata>,
-    path: InstancePath,
-  ): StaticInstancePath | null {
-    const metadataAlongPath = this.getElementInstanceMetadataAlongPath(metadata, path)
-    if (metadataAlongPath == null) {
-      return null
-    } else {
-      // FIXME: Not really a fan, but needs must.
-      if (metadataAlongPath.length + 1 === TP.depth(path)) {
-        const scenePath = TP.scenePathForPath(path)
-        const pathElements = traverseEither<ElementInstanceMetadata, id, ElementInstanceMetadata>(
-          (metadataToTransform) => {
-            const staticUID = Utils.defaultIfNull(
-              metadataToTransform.props['data-uid'],
-              metadataToTransform.props[UTOPIA_ORIGINAL_ID_KEY],
-            )
-            // Should really not fall past this case,
-            // but protecting against it nonetheless.
-            if (staticUID == null) {
-              return left(metadataToTransform)
-            } else {
-              return right(staticUID)
-            }
-          },
-          metadataAlongPath,
-        )
-        return eitherToMaybe(
-          mapEither((elems) => {
-            return TP.staticInstancePath(scenePath, elems)
-          }, pathElements),
-        )
-      } else {
-        return TP.asStatic(path)
-      }
-    }
+  dynamicPathToStaticPath(path: InstancePath): StaticInstancePath | null {
```

Comment:
this is the root of the refactor, I removed the `metadata` parameter, and I agressively removed it from all the places where it was not used anymore